src/runtime-rs/crates/hypervisor/src/device/driver/virtio_vsock.rs

```diff
@@ -121,46 +118,85 @@ const CID_RETRY_COUNT: u32 = 50;
 
 impl VsockDevice {
     pub async fn new(id: String) -> Result<Self> {
-        let vhost_fd = OpenOptions::new()
-            .read(true)
-            .write(true)
-            .open(VHOST_VSOCK_DEVICE)
+        let (guest_cid, _file) = generate_vhost_sock_cid().await?;
+
+        Ok(Self {
+            id,
+            config: VsockConfig { guest_cid },
+        })
+    }
+}
+
+#[async_trait]
+impl Device for VsockDevice {
+    async fn attach(&mut self, h: &dyn hypervisor) -> Result<()> {
+        h.add_device(DeviceType::Vsock(self.clone()))
             .await
-            .context(format!(
-                "failed to open {}, try to run modprobe vhost_vsock.",
-                VHOST_VSOCK_DEVICE
-            ))?;
-        let mut rng = rand::thread_rng();
-
-        // Try 50 times to find a context ID that is not in use.
-        for _ in 0..CID_RETRY_COUNT {
-            // First usable CID above VMADDR_CID_HOST (see vsock(7))
-            let first_usable_cid = 3;
-            let rand_cid = rng.gen_range(first_usable_cid..=(u32::MAX));
-            let guest_cid =
-                unsafe { vhost_vsock_set_guest_cid(vhost_fd.as_raw_fd(), &(rand_cid as u64)) };
-            match guest_cid {
-                Ok(_) => {
-                    return Ok(VsockDevice {
-                        id,
-                        config: VsockConfig {
-                            guest_cid: rand_cid,
-                            vhost_fd,
-                        },
-                    });
-                }
-                Err(nix::Error::EADDRINUSE) => {
-                    // The CID is already in use. Try another one.
-                }
-                Err(err) => {
-                    return Err(err).context("failed to set guest CID");
-                }
-            }
-        }
+            .context("add hybrid vsock device.")?;
```

Comment:
This should read `"add vsock device"`, shouldn't it?

Reply:
Thx @pmores!
Fixed